File: leetcode/best-time-to-buy-and-sell-stock.java
//https://leetcode.com/problems/best-time-to-buy-and-sell-stock/

class Solution {
    public int maxProfit(int[] prices) {  
        int max = 0;
        int min = prices[0];
        
        for(int i = 1; i<prices.length; i++){
            if(min>prices[i]){
                min = prices[i];
            }else if(prices[i]-min > max){
                max = prices[i]-min;
            }
        }
        return max;
    }
}

File: leetcode/best_time_to_buy_and_sell_stock.py
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock/


class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        min_prev: int = 1000000
        max_profit: int = 0
        for i in prices:
            max_profit = max(max_profit, (i - min_prev))
            min_prev = min(min_prev, i)

        return max_profit
